xbmc/filesystem/ISO9660Directory.cpp

```diff
@@ -13,6 +13,7 @@
 #include "utils/URIUtils.h"
 
 #include <cdio++/iso9660.hpp>
+#include "storage/cdioSupport.h"
```

Comment:
Also including cdiosupport/libcdio or media detect components into a bfs implementation is clearly wrong and won't be accepted. There's a reason this is "broken" for 4 years, doing a quick hack to fix the issue is easy, doing it properly and future proof is a completely different story. What about Blu-ray's for example?
This will simply get broken rather sooner than later...